src/server/tiered_storage.cc

```diff
@@ -377,11 +383,9 @@ void TieredStorage::RunOffloading(DbIndex dbid) {
   PrimeTable::Cursor start_cursor{};
 
   // Loop while we haven't traversed all entries or reached our stash io device limit.
-  // Keep number of iterations below resonable limit to keep datastore always responsive
-  size_t iterations = 0;
   do {
     offloading_cursor_ = table.TraverseBySegmentOrder(offloading_cursor_, cb);
-  } while (offloading_cursor_ != start_cursor && stash_limit > 0 && iterations++ < 100);
+  } while (offloading_cursor_ != start_cursor && stash_limit > 0);
```

Comment:
the dilemma is that 100 is not enough to keep the disk busy during the loading phase. My options are finding some reasonable settings for the general case, or calling RunOffloading() from the rdb loader to keep it busy.

Reply:
what about calling Stash directly like you do in SET ?

Reply:
Added, the problem is that when we issue too many stashes, it overflows the limit, so we still rely on background offloading to finish the rest 🤔